src/target/riscv/riscv-013.c

```diff
@@ -2716,21 +2705,43 @@ static uint32_t sb_sbaccess(unsigned int size_bytes)
 	return 0;
 }
 
-static int sb_write_address(struct target *target, target_addr_t address,
-							bool ensure_success)
+static unsigned int get_sbaadress_reg_count(const struct target *target)
 {
 	RISCV013_INFO(info);
-	unsigned int sbasize = get_field(info->sbcs, DM_SBCS_SBASIZE);
+	const unsigned int sbasize = get_field(info->sbcs, DM_SBCS_SBASIZE);
+	assert(sbasize % 32 == 0);
+	return sbasize / 32;
```

Comment:
Detail: I would recommend to not assume that `sbasize` is divisible by 32 as there is no such requirement in the spec. I understand that it is not very likely to happen in practice, but still, it is IMO worth to count with that case:

```suggestion
	assert(DIV_ROUND_UP(sbasize, 32) <= 4);  // if you like - for extra safety
	return DIV_ROUND_UP(sbasize, 32);
```

Reply:
Thanks for the catch!
Addressed. Didn't add the assertion, since IMHO it's better placed in `batch_fill_sb_write_address()` and it's already there:
https://github.com/riscv-collab/riscv-openocd/blob/6d4ad0033ed4e57c1b64998992dcc7a858d7ccd5/src/target/riscv/riscv-013.c#L2725

Reply:
Thanks.